File: api/Models/ProductInBasket.cs
namespace api.Models
{
    public class ProductInBasket
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public Product? Product { get; set; }
        public int BasketId { get; set; }
        public Basket? Basket { get; set; }
        public int Quantity { get; set; }
    }
}

File: webApp/src/app/shared/models/productInBasket.ts
import { Basket } from "./basket";
import { Product } from "./product";

export class ProductInBasket {
    id: number = 0;
    productId: number = 0;
    product: Product = new Product();
    basketId: number = 0;
    basket: Basket = new Basket();
    quantity: number = 0;
    totalPrice: number = 0;
}
